apps/admin-gui/src/app/shared/components/hosts-list/hosts-list.component.ts

```diff
@@ -0,0 +1,109 @@
+import {
+  AfterViewInit,
+  Component,
+  EventEmitter,
+  Input,
+  OnChanges,
+  Output,
+  SimpleChanges,
+  ViewChild
+} from '@angular/core';
+import { Host } from '@perun-web-apps/perun/openapi';
+import { SelectionModel } from '@angular/cdk/collections';
+import { MatPaginator, PageEvent } from '@angular/material/paginator';
+import { MatSort } from '@angular/material/sort';
+import { MatTableDataSource } from '@angular/material/table';
+import { TABLE_ITEMS_COUNT_OPTIONS } from '@perun-web-apps/perun/utils';
+
+@Component({
+  selector: 'perun-web-apps-hosts-list',
+  templateUrl: './hosts-list.component.html',
+  styleUrls: ['./hosts-list.component.css']
+})
+export class HostsListComponent implements AfterViewInit, OnChanges {
+
+  constructor() { }
+
+  @ViewChild(MatSort, { static: true }) set matSort(ms: MatSort) {
+    this.sort = ms;
+    this.setDataSource();
+  }
+
+  @Input()
+  hosts: Host[] = [];
+  @Input()
+  hideColumns: string[] = [];
```

Comment:
I guess you probably inspired in group-table. The thing is this table will be most probably used only in the component you created. And you didn't have any use for this Input() in that component so please remove it.